state-transition/core/state_processor_genesis.go

```diff
@@ -143,5 +150,48 @@ func (sp *StateProcessor[
 		return nil, err
 	}
 
-	return sp.processValidatorsSetUpdates(st)
+	activeVals, err := sp.getActiveVals(st, 0)
+	if err != nil {
+		return nil, err
+	}
+	return sp.validatorSetsDiffs(nil, activeVals), nil
+}
+
+//nolint:lll // let it be.
+func (sp *StateProcessor[
+	_, _, _, BeaconStateT, _, _, _, _, _, _, _, _, ValidatorT, _, _, _, _,
+]) processGenesisActivation(
+	st BeaconStateT,
+) error {
+	switch {
+	case sp.cs.DepositEth1ChainID() == spec.BartioChainID:
+		// nothing to do
+		return nil
+	case sp.cs.DepositEth1ChainID() == spec.BoonetEth1ChainID:
+		// nothing to do
+		return nil
+	default:
+		vals, err := st.GetValidators()
+		if err != nil {
+			return fmt.Errorf("genesis activation, failed listing validators: %w", err)
+		}
+		minEffectiveBalance := math.Gwei(sp.cs.EjectionBalance() + sp.cs.EffectiveBalanceIncrement())
+
+		var idx math.ValidatorIndex
+		for _, val := range vals {
+			if val.GetEffectiveBalance() < minEffectiveBalance {
+				continue
+			}
```

Comment:
just a sanity check for myself: this case should only happen if we include a genesis deposit under min stake? In which case we don't set the activation epoch

Reply:
This is correct. I synced the first 2000 blocks and had no issue, so we should be good